### DISCUSSION TITLE
Discussion: single section switch

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/csharplang#257] [General]
From C++, (https://github.com/dotnet/roslyn/issues/11292)
```cs
switch (specialType)
case SpecialType.System_Byte:
case SpecialType.System_SByte:
case SpecialType.System_Int16:
case SpecialType.System_UInt16:
case SpecialType.System_Int32:
case SpecialType.System_UInt32:
case SpecialType.System_Int64:
case SpecialType.System_UInt64:
    return true;

return false;
```
->
```cs
switch (specialType)
{
    case SpecialType.System_Byte:
    case SpecialType.System_SByte:
    case SpecialType.System_Int16:
    case SpecialType.System_UInt16:
    case SpecialType.System_Int32:
    case SpecialType.System_UInt32:
    case SpecialType.System_Int64:
    case SpecialType.System_UInt64:
        return true;

    default:
        return false;
}
```
The single section switch doesn't need `break` and it would belong to an enclosing statement (if any).

**Comment by iam3yal** [comment]
I think that the parenthesis are useful here for readability, not sure why people want to get rid of them but not needing a `break` is useful but then again wouldn't the `match` expression cover this case too?

**Comment by alrz** [comment]
@eyalsk match cannot be used as an statement (except with a discard if it doesn't return void). Not requiring break here is because there is only a single section so the control will fall out of switch anyways.

**Comment by DavidArno** [comment]
@alrz,

Assuming you are thinking just of `return` patterns here, then the `match` feature should cover this:
```cs
return specialType match (
    case SpecialType.System_Byte:
    case SpecialType.System_SByte:
    case SpecialType.System_Int16:
    case SpecialType.System_UInt16:
    case SpecialType.System_Int32:
    case SpecialType.System_UInt32:
    case SpecialType.System_Int64:
    case SpecialType.System_UInt64: true
    case * : false
    );
```

Alternatively, with an `or` pattern:
```cs
return specialType match (
    case SpecialType.System_Byte or
         SpecialType.System_SByte or
         SpecialType.System_Int16 or
         SpecialType.System_UInt16 or
         SpecialType.System_Int32 or
         SpecialType.System_UInt32 or
         SpecialType.System_Int64 or
         SpecialType.System_UInt64: true
    case * : false
    );
```

**Comment by alrz** [comment]
@DavidArno Assuming that you are not returning a value
```cs
e match { .. }; // ERROR
```
Even with a discard, you can't use void returning methods,
```cs
_ = e match { /* .. VoidMethod() .. */ };
```
Though that would probably work because of https://github.com/dotnet/roslyn/issues/17674 :smile:

**Comment by iam3yal** [comment]
@alrz Got'cha, the only issue I have with this is mostly syntactic. :)

**Comment by alrz** [comment]
@eyalsk 

This *is* just syntactic. I really hope that `match` gets a statement form, even if it needs some special casing.

**Comment by iam3yal** [comment]
@alrz Yeah I realized that but I said that because I don't want to lose the parenthesis, I know it's subtle but for me, I'm used to having them everywhere so yup I'd really love the match expression to have a statement form over this. :)

_Feels like I'm ranting over nothing._ 😄
